main.tf

```diff
@@ -1 +1,62 @@
-# Your Terraform code goes here :)
+locals {
+  policy_name = length(var.policy_name) > 0 ? var.policy_name : "${var.base_name}_policy"
+}
+
+module "lacework_oci_credentials" {
+  source        = "lacework/iam-user/oci"
+  version       = ">= 0.2.0"
+  tenancy_id    = var.tenancy_id
+  create        = var.create
+  freeform_tags = var.freeform_tags
+  email         = var.user_email
+  base_name     = var.base_name
+  user_name     = var.user_name
+  group_name    = var.group_name
+}
+
+resource "oci_identity_policy" "lacework_policy" {
+  count          = var.create ? 1 : 0
+  compartment_id = var.tenancy_id
+  description    = "Permission policy for the Lacework CSPM user"
+  name           = local.policy_name
+  freeform_tags  = var.freeform_tags
+  statements = [
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect compartments in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to read buckets in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect volumes in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect security-lists in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to read users in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect groups in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect policies in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect domains in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect tag-defaults in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect instances in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect subnets in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect network-security-groups in tenancy",
+  ]
```

Comment:
@fdong-lw @akshara-lw Can you take a look at this set of permissions needed for Lacework configuration integration. Is this right as a minimal set of permissions?

Reply:
right, those support first 15 APIs for private preview. We can use it as start, but it could be more within coverage. https://docs.google.com/spreadsheets/d/1PLVmmE3yy7WIo5U3rxh36mN7A6bBYtzVPzs6g8m6o0I/edit?pli=1#gid=1516278081